### PR TITLE
Add big data donation banners and new settings form.

### PR DESCRIPTION
**Pull request by cbwebdevelopment**
Provides a new UI for setting up data donation for logged in patients.
Details at https://trello.com/c/v8ovj43d

NOTE: there is a temporary fix in `hydrophone` that I've put in place in order to fix a bug where you could not re-add previously added and removed donation partners.  The temp fix can be found on the `clintb/bdd-ui-updates` branch in that repo.  There will be an official PR at some point coming from the backend team at Tidepool, but this fix will suffice for proceeding locally with the review.
https://github.com/tidepool-org/hydrophone/tree/clintb%2Fbdd-ui-updates

Due to the upcoming hydrophone PR, it may be the case that there will need to be small changes to this PR, depending on how the backend team approaches the solution.  

I don't think we need to hold off on code review, however, as I expect any changes would be very minor, and the bulk of this PR would be unaffected.

### CODE REVIEW

**Comment by cbwebdevelopment** [comment]
[Share with Big Data Donation UI on Share page](https://trello.com/c/v8ovj43d/192-share-with-big-data-donation-ui-on-share-page)

File: app/redux/reducers/misc.js [hunk 1]
```diff
@@ -461,3 +480,33 @@ export const pendingReceivedInvites = (state = initialState.pendingReceivedInvit
       return state;
   }
 };
+
+export const dataDonationAccounts = (state = initialState.dataDonationAccounts, action) => {
```
Comment (krystophv) [inline app/redux/reducers/misc.js]: It looks like these `misc` reducers slipped through testing. (not that I can tell that there's anything amiss, but it's probably better to have them also unit tested)

Comment (cbwebdevelopment) [inline app/redux/reducers/misc.js]: Done.  Thanks for catching this.

File: test/unit/redux/reducers/showingDonateBanner.test.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      expect(intermediate).to.be.true;
+
+      let nextState = reducer(null, action);
+
+      expect(nextState).to.be.true;
+    });
+  });
+
+  describe('hideDonateBanner', () => {
+    it('should set state to false', () => {
```
Comment (krystophv) [inline test/unit/redux/reducers/showingDonateBanner.test.js]: It's a nitpick, but should be `set state to null`?

Comment (cbwebdevelopment) [inline test/unit/redux/reducers/showingDonateBanner.test.js]: Doh!  Yeah, valid nitpick 👍

**Comment by krystophv** [comment]
Cool - LGTM 👍

File: test/unit/redux/reducers/showingDonateBanner.test.js (hunk 2 above)
Comment (cbwebdevelopment) [inline test/unit/redux/reducers/showingDonateBanner.test.js]: Fixed.

**Pull request merged**